src/tools/clippy/tests/ui/needless_match.fixed

```diff
@@ -1,7 +1,7 @@
 #![warn(clippy::needless_match)]
 #![allow(clippy::manual_map)]
 #![allow(dead_code)]
-
+#![allow(unused)]
```

Comment:
NIT: Isn't `unused` allowed by default in the `ui_test` framework? Why is this explicit allow necessary here? :thinking: